### PR TITLE
Add DomainValidator

### PR DESCRIPTION
**Pull request by toban**
Rip out part of the validation logic in the WikiController to it's own
helper/service class. Adds ForbiddenSubdomainRule that takes a list of
words and does simple validation against that.

- This does not check for profanity, but should be straight forward to add

### CODE REVIEW

File: app/Helper/DomainValidator.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public function validate( $domain ) {
+
+        $isSubdomain = WikiController::isSubDomain( $domain, $this->subDomainSuffix );
+
+        if ($isSubdomain) {
+            $subDomainSuffixLength = strlen($this->subDomainSuffix);
+            $requiredSubdomainPrefixChars = 5;
+            // We want at least 5 chars for the site sub domain
+            // This also stops things like mail. www. pop. ETC...
+            // TODO add a real disallow list for various wanted subdomains
```
Comment (toban) [inline app/Helper/DomainValidator.php:30]:
```suggestion
```

File: app/Rules/ForbiddenSubdomainRule.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Rules;
```
Comment (tarrow) [inline app/Rules/ForbiddenSubdomainRule.php:3]: This looks really nice and reads (to me) as nice idiomatic Laravel :)

File: app/Rules/ForbiddenSubdomains.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    'css3',
+    'dashboard',
+    'demo',
+    'dev',
+    'developer',
+    'developers',
+    'development',
+    'diff',
+    'docs',
+    'donate',
```
Comment (tarrow) [inline app/Rules/ForbiddenSubdomains.php:59]:
```suggestion
    'donate',
    'dyna',
```

**Review by tarrow — APPROVED** [review]

File: app/Rules/ForbiddenSubdomains.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (tarrow) [reply app/Rules/ForbiddenSubdomains.php:59]: done

**Pull request merged**